azure_functions_worker/utils/dependency.py

```diff
@@ -112,24 +123,50 @@ def prioritize_customer_dependencies(cls):
         As for Linux Consumption, this will only remove worker_deps_path,
         but the customer's path will be loaded in function_environment_reload.
 
-        The search order of a module name in customer frame is:
+        The search order of a module name in customer's paths is:
         1. cx_deps_path
-        2. cx_working_dir
+        2. worker_deps_path
+        3. cx_working_dir
         """
+        # Try to get the latest customer's working directory
+        # cx_working_dir => cls.cx_working_dir => AzureWebJobsScriptRoot
+        working_directory: str = ''
+        if cx_working_dir:
+            working_directory: str = os.path.abspath(cx_working_dir)
+        if not working_directory:
+            working_directory = cls.cx_working_dir
+        if not working_directory:
+            working_directory = os.getenv(AZURE_WEBJOBS_SCRIPT_ROOT, '')
+
+        # Try to get the latest customer's dependency path
+        cx_deps_path: str = cls._get_cx_deps_path()
+        if not cx_deps_path:
+            cx_deps_path = cls.cx_deps_path
+
+        logger.info('Applying prioritize_customer_dependencies:'
+                    f' worker_dependencies: {cls.worker_deps_path},'
+                    f' customer_dependencies: {cx_deps_path},'
+                    f' working_directory: {working_directory}')
+
         cls._remove_from_sys_path(cls.worker_deps_path)
         cls._add_to_sys_path(cls.cx_deps_path, True)
-        cls._add_to_sys_path(cls.cx_working_dir, False)
 
         # Deprioritize worker dependencies but don't completely remove it
         # Otherwise, it will break some really old function apps, those
         # don't have azure-functions module in .python_packages
         # https://github.com/Azure/azure-functions-core-tools/pull/1498
         cls._add_to_sys_path(cls.worker_deps_path, False)
 
-        logger.info(f'Start using customer dependencies {cls.cx_deps_path}')
+        # The modules defined in customer's working directory should have the
+        # least priority since we uses the new folder structure.
+        # Please check the "Message to customer" section in the following PR:
+        # https://github.com/Azure/azure-functions-python-worker/pull/726
```

Comment:
Can we make a pinned issue as well (or check if this is added to Python Dev Docs itself)?